src/utils.rs

```diff
@@ -81,53 +125,171 @@ pub fn find_message_type_in_file_descriptor(
   descriptor: &FileDescriptorProto
 ) -> anyhow::Result<DescriptorProto> {
   descriptor.message_type.iter()
-    .find(|message| message.name.clone().unwrap_or_default() == message_name)
+    .find(|message| message.name() == message_name)
     .cloned()
     .ok_or_else(|| anyhow!("Did not find a message type '{}' in the file descriptor '{}'",
       message_name, descriptor.name.as_deref().unwrap_or("unknown")))
 }
 
-/// Finds message descriptor in a map of file descriptors.
-pub fn find_message_descriptor_from_hash_map(
-  message_name: &str,
-  package: Option<&str>,
+// TODO: handle nested types properly
+// current name resolution is dumb - just splits package and message name by a dot
+// but if you have .package.Message.NestedMessage.NestedMessageDeeperLevel this whole structure breaks down
+// because we'll be looking for packages with the name `.package.Message.NestedMessage`
+// while here the package is `.package` and then we need to find message called `Message` and go over it's nested types.
```

Comment:
It's not, that's why it's something to look into at a later time. I think the next step would be nested types with fully qualified names, and I don't think we should tackle relative names any time soon - at least currently all the descriptors I've seen had fully-qualified type names filled in, I wonder if this can be adjusted by protoc flags? Because the docs on descriptor types mention a possibility of relative names.

Reply:
I wonder if there's a way to confirm that descriptors we get from `protoc` will always have fully-qualified type names filled in. If they do, this makes our life so much easier, this means all the heavy-lifting of resolving relative references and all we have to do is to figure out if the dot-separated token refers to a package or a type, which is annoying but not too hard to do